092-self-healing-using-cruise-control.md

```diff
@@ -0,0 +1,300 @@
+  reason: Anomaly was detected in the cluster GOAL_VIOLATION # Detected Anomaly type
+  related: 
+    kind: Pod
+    name: <cruise-control-pod-name>
+    namespace: <project-namespace>
+  reportingComponent: cruise-control
+  reportingInstance: cruise-control
+  source: {}
+  type: Normal
+```
+
+### Strimzi Operations while self-healing is running/fixing an anomaly 
+
+#### What happens when self-healing is running/fixing an anomaly and brokers are rolled
+
+When an anomaly is detected, Cruise Control may try to fix/mitigate it by moving the partition replicas across the brokers in the Kafka cluster.
+If, during the process of moving partitions replicas among the brokers in the Kafka cluster, some broker that was being used gets deleted or rolled, then Cruise Control would finish self-healing process and log that `self-healing finished successful with error`.
+Since the self-healing process was not completed successfully, it would start again with a new anomalyId to be fixed properly.
+
+#### Self-healing behaviour with KafkaRebalance 
+
+* If self-healing is ongoing and a `KafkaRebalance` resource is created/applied in the middle of it then Cruise Control will still generate an optimization proposal.
+The proposal should be based on the cluster shape at that particular timestamp.
+* If self-healing is ongoing and the generated proposal is `approved` then the `KafkaRebalance` resource will move to `NotReady` state, stating that there is already a rebalance ongoing.
+
+For example:
+
+```yaml
+status:
+  conditions:
+  - lastTransitionTime: "2025-01-08T08:24:01.314273452Z"
+    message: 'Error for request: my-cluster-cruise-control.myproject.svc:9090/kafkacruisecontrol/rebalance?json=true&dryrun=false&verbose=true&skip_hard_goal_check=false&rebalance_disk=false.
+      Server returned: Error processing POST request ''/rebalance'' due to: ''java.lang.IllegalStateException:
+      Cannot start a new execution while there is an ongoing execution. Please use
+      stop_ongoing_execution=true to stop ongoing execution and start a new one.''.'
+    reason: CruiseControlRestException
+    status: "True"
+    type: NotReady
+  - lastTransitionTime: "2025-01-08T08:24:01.349653295Z"
+    message: 'Wrong annotation value: approve. The rebalance resource supports the
+      following annotations in NotReady state: [refresh]'
+    reason: InvalidAnnotation
+    status: "True"
+    type: Warning
+  observedGeneration: 1
+```
+
+* In case, a manual rebalance is ongoing and self-healing is triggered, then the self-healing process waits till the rebalancing is over.
+* If the KafkaRebalance is generated again using the `refresh` annotation in between of self-healing, then a proposal is still generated based on the cluster shape at that particular timestamp.
+
+## Affected/not affected projects
+
+This change will affect the Strimzi cluster operator and a new repository named `kubernetes-event-notifier` will be added under the Strimzi organisation. 
```

Comment:
It should probably include the `cruise-control` somewhere in the name? Please also move the information abotu the new repository to the section about the notifier (unless you remove it as I suggest of course). You should also clarify there how it will be distributed (likely through Maven Central?).